test/test_differentiation.py

```diff
@@ -418,3 +418,12 @@ def test_centered_difference_noaxis_vs_axis(data_2d_resolved_pde):
             slow_differences_t,
             atol=atol,
         )
+
+
+def test_multidimensional_differentiation():
+    X = np.random.random(size=(10, 100, 2))
+    t = np.arange(0, 10, 0.1)
+
+    X_dot = SINDyDerivative(kind="kalman", axis=-2)._differentiate(X, t)
+
+    assert X_dot.shape == X.shape
```

Comment:
Let's make sure we differentiate the correct axis, rather than just verifying that it runs without error.  E.g.:

* create a short arange of ints
* tile it /repeat it to make it 3D, so that x[i,i,:] is the original arange
* differentiate it (use something like finite difference, so that we know the expected result)
* test that every slice is what we expect (ones)

Change the name to `test_nd_diff_sindy_derivative()` to reflect that we're only testing one class.

Reply:
It's also worth checking whether this preserves `AxesArray` types.

Reply:
Instead of just `SINDyDerivative`, the test is written for `FiniteDifference`, `SmoothedFiniteDifference` and `SpectralDerivative` as well.